ex03.exs

```diff
@@ -51,14 +51,77 @@ defmodule Ex03 do
   """
 
   def pmap(collection, process_count, function) do
-    # your code goes here
+    collection
+      |> splitIntoChunks(process_count)
+      |> convertToProcesses(function)
+      |> runProcesses()
+
     # I'm hoping to see a simple pipeline as the body of this function...
   end
 
-  # and here...
+  def splitIntoChunks(collection, numProcesses) do
+    countElements = Enum.count(collection) / numProcesses
+      |> Kernel.trunc()
+    Enum.chunk_every(collection, countElements, countElements,[])
+      |> ensureCorrectNumChunks(numProcesses)
+  end
+
+  def ensureCorrectNumChunks(collection, numProcesses) do
+    if length(collection) != numProcesses do
+      {leftover, _restCollection} = List.pop_at(collection, numProcesses - 1)
+      List.update_at(collection, numProcesses - 1, &(List.insert_at(&1, numProcesses, leftover)))
+    end
+    collection
+  end
```

Comment:
This function isn't needed at all if you just change `Enum.count(collection) / numProcesses |> Kernel.trunc()` to

    div(Enum.count(collection) + numProcesses - 1), numProcesses)